frankenstein/distributor.go

```diff
@@ -195,3 +245,28 @@ func (d *Distributor) LabelValuesForLabelName(ctx context.Context, labelName mod
 func (*Distributor) NeedsThrottling(_ context.Context) bool {
 	return false
 }
+
+// Describe implements prometheus.Collector.
+func (d *Distributor) Describe(ch chan<- *prometheus.Desc) {
+	d.queryDuration.Describe(ch)
+	ch <- d.consulUpdates.Desc()
+	ch <- d.receivedSamples.Desc()
+	d.sendDuration.Describe(ch)
+	ch <- numClientsDesc
+}
+
+// Collect implements prometheus.Collector.
+func (d *Distributor) Collect(ch chan<- prometheus.Metric) {
+	d.queryDuration.Collect(ch)
+	ch <- d.consulUpdates
+	ch <- d.receivedSamples
+	d.sendDuration.Collect(ch)
+
+	d.clientsMtx.RLock()
+	defer d.clientsMtx.RUnlock()
+	ch <- prometheus.MustNewConstMetric(
+		numClientsDesc,
+		prometheus.GaugeValue,
+		float64(len(d.clients)),
+	)
+}
```

Comment:
I've always wondered, why do it like this and not register them to the global list of metrics?